As we use DEB_CONFIG_INSTALL_DIR ?= /usr/share/kde4/config
and we want kdm configuration files in /etc/kde4/kdm,
Let's hardcode it and introduce KDM_CONFDIR.

--- a/config-workspace.h.cmake
+++ b/config-workspace.h.cmake
@@ -123,6 +123,9 @@
 /* KDE's static data directory */
 #define KDE_DATADIR "${DATA_INSTALL_DIR}"
 
+/* KDM's configuration directory */
+#define KDM_CONFDIR "/etc/kde4"
+
 /* Define where your java executable is */
 #undef PATH_JAVA
 
--- a/kcontrol/kdm/background.cpp
+++ b/kcontrol/kdm/background.cpp
@@ -43,9 +43,9 @@
 		" If it is disabled, you have to look after the background yourself."
 		" This is done by running some program (possibly xsetroot) in the script"
 		" specified in the Setup= option in kdmrc (usually Xsetup).") );
-	m_simpleConf = KSharedConfig::openConfig(
-		config->group( "X-*-Greeter" ).readEntry(
-			"BackgroundCfg", KDE_CONFDIR "/kdm/backgroundrc" ) );
+	m_simpleConf = KSharedConfig::openConfig( 
+	    config->group( "X-*-Greeter" ).readEntry( 
+		    "BackgroundCfg", KDM_CONFDIR "/kdm/backgroundrc" ) );
 	m_background = new BGDialog( this, m_simpleConf );
 
 	connect( m_background, SIGNAL(changed( bool )), SIGNAL(changed()) );
--- a/kcontrol/kdm/main.cpp
+++ b/kcontrol/kdm/main.cpp
@@ -187,8 +187,8 @@
 	for (tgmapci = tgmap.begin(); tgmapci != tgmap.end(); ++tgmapci)
 		kWarning() << "user(s) '" << tgmapci.value().join( "," )
 		<< "' have unknown GID " << tgmapci.key() << endl;
-
-	config = new KConfig( QString::fromLatin1(KDE_CONFDIR "/kdm/kdmrc"), KConfig::SimpleConfig );
+    
+	config = new KConfig( QString::fromLatin1(KDM_CONFDIR "/kdm/kdmrc"), KConfig::SimpleConfig);
 
 	QVBoxLayout *top = new QVBoxLayout( this );
 	top->setMargin( 0 );
--- a/kdm/config.def
+++ b/kdm/config.def
@@ -29,7 +29,7 @@
 #define RCVERMAJOR 2
 #define RCVERMINOR 4
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF KDM_CONFDIR "/kdm"
 #define KDMDATA KDE_DATADIR "/kdm"
 
 #ifdef _AIX
--- a/kdm/kfrontend/kdmctl.c
+++ b/kdm/kfrontend/kdmctl.c
@@ -150,7 +150,7 @@
 {
 	char *dpy = getenv( "DISPLAY" );
 	const char *ctl = getenv( "DM_CONTROL" );
-	const char *cfg = KDE_CONFDIR "/kdm/kdmrc";
+	const char *cfg = KDM_CONFDIR "/kdm/kdmrc";
 	char *ptr;
 	int fd;
 
--- a/kdm/kfrontend/genkdmconf.c
+++ b/kdm/kfrontend/genkdmconf.c
@@ -2766,6 +2766,7 @@
 
 
 static const char *oldkdes[] = {
+	KDM_CONFDIR,
 	KDE_CONFDIR,
 	"/opt/kde4/share/config",
 	"/usr/local/kde4/share/config",
@@ -2774,6 +2775,7 @@
 	"/usr/local/kde/share/config",
 	"/usr/local/share/config",
 	"/usr/share/config",
+	"/etc/kde3",
 
 	"/opt/kde3/share/config",
 	"/usr/local/kde3/share/config",
@@ -2817,7 +2819,7 @@
 "    Note that you possibly need to use --no-old-kde to make this take effect.\n"
 "  --old-kde /path/to/old/kde-config-dir\n"
 "    Where to look for the kdmrc of a previously installed kdm.\n"
-"    Default is to scan " KDE_CONFDIR " and\n"
+"    Default is to scan " KDM_CONFDIR " and\n"
 "    {/usr,/usr/local,{/opt,/usr/local}/{kde4,kde,kde3}}/share/config.\n"
 "  --no-old\n"
 "    Do not look at older xdm/kdm configurations, just create default config.\n"
